feature/KAN-32/redux-toolkit

YelyzavetaDemchenko opened the pull request:
This summarizes work done so far adding redux-toolkit.

Implemented ProductSlice, store, added redux-persist for work with local storage. Synchronized product counts between home page and Basket.

YelyzavetaDemchenko committed:
KAN-8-29/ fixed voucher bug

YelyzavetaDemchenko committed:
KAN-30/sync with basket and dicounts

Adds discounts on product card and synchronizes it with basket and vouchers

YelyzavetaDemchenko committed:
KAN-30/API

Integrated API discounts

YelyzavetaDemchenko committed:
feature/KAN-32/redux-toolkit

Added ProductSlice, selectors, reducers and operations to get products from API and add product to Basket.

YelyzavetaDemchenko committed:
feature/KAN-32/redux-toolkit

Synchronized the product add to the Basket and set counters on product card

VolodymyrKholiavko commented on src/modules/Basket/InfoToPay.tsx:
```diff
@@ -4,7 +4,7 @@ import { useBasketContext } from '../../contexts/BasketContext'
   const { totalPrice, totalWeight, voucher } = useBasketContext()
   const discountedPrice = totalPrice * voucher.discount
-  const isDiscounted = voucher.discount !== 1
+  const isVoucherActive = totalPrice !== 0 && voucher.discount !== 1
```
ternar checks should be used inside return( )  only, here's better implement if/else/switch

VolodymyrKholiavko commented on src/modules/Basket/ProductsList/ProductListItem.tsx:
```diff
@@ -1,33 +1,44 @@
+
+  const isDiscounted = Boolean(item.product.discount)
+
+  const price = isDiscounted
```
ternar checks should be used inside return only

VolodymyrKholiavko commented on src/modules/Basket/ProductsList/ProductListItem.tsx:
```diff
@@ -1,33 +1,44 @@
           <Text fontSize={13}>
-            {Number(product.weight * product.count).toFixed(2)} gram /{' '}
-            {product.size * product.count} шт.
+            {Number(item.product.weight * item.count).toFixed(2)} gram /{' '}
```
suggestion: calculating also better use in body of component (between states and render (return( ) )

VolodymyrKholiavko reviewed (changes requested):
good

YelyzavetaDemchenko committed:
KAN-32/product btn, Basket info and continue btn

Adds blocking Product btn when it is added to basket, Basket info to pay and fixed continue btn ih the Basket

chernovsushki committed:
2204 task about incr decr delete in selected pruducts 220424 1924

YelyzavetaDemchenko committed:
Merge pull request #5 from NeptunesProject/feature/KAN-32/redux-toolkit-chernov

2204 task about incr decr delete in selected pruducts 220424 1924

chernovsushki committed:
fixed/erasing cart after order sending 1241230424

YelyzavetaDemchenko committed:
Merge pull request #6 from NeptunesProject/feature/KAN-32/redux-toolkit-chernov-2304

Feature/kan 32/redux toolkit chernov 2304

YelyzavetaDemchenko committed:
KAN-32/basket

Remade on redux basket additional info btns(person, sticks, studySticks), added them to locale storage.
